simplified-accessibility/src/main/java/org/nypl/simplified/accessibility/AccessibilityService.kt

```diff
@@ -0,0 +1,159 @@
+package org.nypl.simplified.accessibility
+
+import android.accessibilityservice.AccessibilityServiceInfo.FEEDBACK_SPOKEN
+import android.content.Context
+import android.content.Context.ACCESSIBILITY_SERVICE
+import android.view.accessibility.AccessibilityManager
+import androidx.annotation.UiThread
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.OnLifecycleEvent
+import io.reactivex.disposables.CompositeDisposable
+import org.nypl.simplified.books.book_registry.BookRegistryReadableType
+import org.nypl.simplified.books.book_registry.BookStatus
+import org.nypl.simplified.books.book_registry.BookStatusEvent
+import org.nypl.simplified.ui.thread.api.UIThreadServiceType
+
+/**
+ * The default accessibility service.
+ */
+
+class AccessibilityService private constructor(
+  private val accessibilityManager: AccessibilityManager,
+  private val bookRegistry: BookRegistryReadableType,
+  private val strings: AccessibilityStringsType,
+  private val toasts: AccessibilityToastsType,
+  private val uiThread: UIThreadServiceType,
+) : AccessibilityServiceType {
+
+  companion object {
+    fun create(
+      context: Context,
+      bookRegistry: BookRegistryReadableType,
+      uiThread: UIThreadServiceType,
+      strings: AccessibilityStringsType = AccessibilityStrings(context.resources),
+      toasts: AccessibilityToastsType = AccessibilityToasts(context)
+    ): AccessibilityServiceType {
+      val accessibilityManager =
+        context.getSystemService(ACCESSIBILITY_SERVICE) as AccessibilityManager
+      return AccessibilityService(
+        accessibilityManager = accessibilityManager,
+        bookRegistry = bookRegistry,
+        strings = strings,
+        toasts = toasts,
+        uiThread = uiThread
+      )
+    }
+  }
+
+  private fun isSpokenFeedbackEnabled(): Boolean {
+    return this.accessibilityManager
+      .getEnabledAccessibilityServiceList(FEEDBACK_SPOKEN)
+      .isNotEmpty()
+  }
+
+  override val spokenFeedbackEnabled: Boolean
+    get() = this.isSpokenFeedbackEnabled()
+
+  @Volatile
+  private var lifecycleOwner: LifecycleOwner? = null
+  private val subscriptions = CompositeDisposable()
+
+  @OnLifecycleEvent(Lifecycle.Event.ON_RESUME)
+  override fun onViewAvailable(owner: LifecycleOwner) {
+    this.lifecycleOwner = owner
+    this.subscriptions.add(this.bookRegistry.bookEvents().subscribe(this::onBookEvent))
+  }
+
+  @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
+  override fun onViewUnavailable(owner: LifecycleOwner) {
+    this.lifecycleOwner = owner
+    this.subscriptions.dispose()
```

Comment:
For future reference: This is a bug.

Why? Because `dispose()` will mark `subscriptions` as disposed, and any subsequent calls to `add()` will silently be ignored.

Reply:
Aha, good catch!